peps/pep-0774.rst

```diff
@@ -0,0 +1,255 @@
+PEP: 774
+Title: Removing the LLVM requirement for JIT builds
+Author: Savannah Ostrowski <savannah@python.org>
+Status: Draft
+Type: Standards Track
+Created: 26-Jan-2025
+Python-Version: 3.14
+
+Abstract
+========
+
+Since Python 3.13, CPython has been able to be configured and built with an
+experimental just-in-time (JIT) compiler via the ``--enable-experimental-jit``
+flag on Linux and Mac and ``--experimental-jit`` on Windows. To build CPython with
+the JIT enabled, users are required to have LLVM installed on their machine
+(initially, with LLVM 16 but more recently, with LLVM 19). LLVM is responsible
+for generating stencils that are essential to our copy-and-patch JIT (see :pep:`744`).
+
+This PEP proposes removing the LLVM build-time dependency for JIT-enabled builds
+by hosting the generated stencils in the CPython repository to simplify the
+contributor experience and address concerns raised at the Python Core Developer
+Sprint in September 2024. That said, there is a clear tradeoff to consider, as
+improved developer experience does come at the cost of increased repository
+size.
+
+It is important to note that this PEP is not a proposal to accept or reject the
+JIT itself but rather to determine whether the build-time dependency on LLVM is
+acceptable for JIT builds moving forward. If this PEP is rejected, we will
+proceed with the status quo, retaining the LLVM build-time requirement. While
+this dependency has served the JIT development process effectively thus far, it
+introduces setup complexity and additional challenges that this PEP seeks to
+alleviate.
+
+Motivation
+==========
+
+At the Python Core Developer Sprint that took place in September 2024, there was
+discussion about the next steps for the JIT - a related discussion also took
+place on `GitHub <https://github.com/python/cpython/issues/115869>`__. As part
+of that discussion, there was also a clear appetite for removing the LLVM
+requirement for JIT builds in preparation for shipping the JIT off by default in
+3.14. The consensus at the sprint was that it would be sufficient to provide
+pre-generated stencils for non-debug builds for Tier 1 platforms and that
+checking these files into the CPython repo would be adequate for the limited
+number of platforms (though more options have been explored; see `Rejected
+Ideas <PEP 774 Rejected Ideas_>`__).
```

Comment:
Headers are automatically reference targets within the document, so you should be able to simplify this (pt1)
```suggestion
number of platforms (though more options have been explored; see `Rejected
Ideas`_).
```